corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -287,6 +301,30 @@ def rebuild_from_transactions(self, transactions, rebuild_transaction, unarchive
         if not self.case.modified_on:
             self.case.modified_on = rebuild_transaction.server_date
 
+    def reconcile_transactions_if_necessary(self, xform):
+        if not self.case.check_transaction_order():
+            try:
+                self.reconcile_transactions(rebuild=True)
+            except ReconciliationError:
```

Comment:
this should probably notify via a soft assert

Reply:
70be6e9be2